pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -75,6 +76,17 @@ type ControllerOptions struct {
 	WatchedNamespace string
 	ResyncPeriod     time.Duration
 	DomainSuffix     string
+
+	// ClusterID identifies the remote cluster in a multicluster env.
+	ClusterID string
```

Comment:
VNID

Reply:
Same. This is actually intended to represent the cluster - k8s or VM cluster or whatever. 
A cluster may or may not be on a separate network, or have multiple networks - and that will be 
handled in the split-horizon PR.